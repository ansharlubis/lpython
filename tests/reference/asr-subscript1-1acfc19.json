{
    "basename": "asr-subscript1-1acfc19",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/subscript1.py",
    "infile_hash": "5d229893d3e13ea4463e8ed47eb3798be0b8c28f5ef6b6c773e87b80",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-subscript1-1acfc19.stdout",
    "stdout_hash": "fa2dc3dabbf7d17201c4bb54fa4fe94a4c702731a110be4791fec18a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}