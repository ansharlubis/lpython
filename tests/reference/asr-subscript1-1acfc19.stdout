([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         test_subscript: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     A: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           A [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     B: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           B [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 [
                              (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                                 ([1m[35mInteger[39m[0m 4 [])) 
                                 ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                                    Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 [])) 
                                    ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                                    ([1m[35mInteger[39m[0m 4 []))))]) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     i: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           i [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     s: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           s [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mCharacter[39m[0m 1 -2 () []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               test_subscript [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringConstant[39m[0m "abc" 
                     ([1m[35mCharacter[39m[0m 1 3 () [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringItem[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-1[39m 
                           ([1m[35mInteger[39m[0m 4 []))) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m88[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-4[39m 
                           ([1m[35mInteger[39m[0m 4 []))) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m89[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-89[39m 
                           ([1m[35mInteger[39m[0m 4 []))) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-3[39m 
                           ([1m[35mInteger[39m[0m 4 []))) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-3[39m 
                        ([1m[35mInteger[39m[0m 4 []))) 
                     ([1m[35mIntegerBinOp[39m[0m 
                        ([1m[35mIntegerUnaryMinus[39m[0m ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m-3[39m 
                           ([1m[35mInteger[39m[0m 4 []))) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ms[39m) 
                  ([1m[35mStringSection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33ms[39m) 
                     ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        Add[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mInteger[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mCharacter[39m[0m 1 -2 () []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mi[39m) 
                  ([1m[35mArrayItem[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mA[39m) [
                     (() ([1m[35mIntegerConstant[39m[0m [36m0[39m 
                        ([1m[35mInteger[39m[0m 4 [])) ())] 
                     ([1m[35mInteger[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mB[39m) 
                  ([1m[35mArraySection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mA[39m) [
                     (([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                        ([1m[35mInteger[39m[0m 4 [])) ())] 
                     ([1m[35mInteger[39m[0m 4 [
                        (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                              ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                              Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                              ([1m[35mInteger[39m[0m 4 [])) 
                              ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                              ([1m[35mInteger[39m[0m 4 []))))]) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mB[39m) 
                  ([1m[35mArraySection[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mA[39m) [
                     (([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                        ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                        ([1m[35mInteger[39m[0m 4 [])))] 
                     ([1m[35mInteger[39m[0m 4 [
                        (([1m[35mIntegerConstant[39m[0m [36m0[39m 
                           ([1m[35mInteger[39m[0m 4 [])) 
                           ([1m[35mIntegerBinOp[39m[0m ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                              ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                              Sub[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                              ([1m[35mInteger[39m[0m 4 [])) 
                              ([1m[35mInteger[39m[0m 4 []) ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                              ([1m[35mInteger[39m[0m 4 []))))]) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [])
(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_subscript: (Subroutine (SymbolTable 2 {A: (Variable 2 A Local () () Default (Integer 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 5 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 4 (Integer 4 []))))]) Source Public Required .false.), B: (Variable 2 B Local () () Default (Integer 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 2 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 1 (Integer 4 []))))]) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), s: (Variable 2 s Local () () Default (Character 1 -2 () []) Source Public Required .false.)}) test_subscript [] [(= (Var 2 s) (StringConstant "abc" (Character 1 3 () [])) ()) (= (Var 2 s) (StringItem (Var 2 s) (IntegerBinOp (IntegerConstant 0 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerBinOp (IntegerConstant 1 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerUnaryMinus (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant -1 (Integer 4 []))) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 2 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerBinOp (IntegerConstant 1 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 88 (Integer 4 [])) (IntegerBinOp (IntegerConstant 1 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerConstant 1 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (IntegerBinOp (IntegerUnaryMinus (IntegerConstant 4 (Integer 4 [])) (Integer 4 []) (IntegerConstant -4 (Integer 4 []))) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerBinOp (IntegerUnaryMinus (IntegerConstant 89 (Integer 4 [])) (Integer 4 []) (IntegerConstant -89 (Integer 4 []))) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 4 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerBinOp (IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) (IntegerBinOp (IntegerUnaryMinus (IntegerConstant 3 (Integer 4 [])) (Integer 4 []) (IntegerConstant -3 (Integer 4 []))) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (Character 1 -2 () []) ()) ()) (= (Var 2 s) (StringSection (Var 2 s) (IntegerBinOp (IntegerConstant 2 (Integer 4 [])) Add (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 3 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])) (Character 1 -2 () []) ()) ()) (= (Var 2 i) (ArrayItem (Var 2 A) [(() (IntegerConstant 0 (Integer 4 [])) ())] (Integer 4 []) ()) ()) (= (Var 2 B) (ArraySection (Var 2 A) [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 3 (Integer 4 [])) ())] (Integer 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 5 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 4 (Integer 4 []))))]) ()) ()) (= (Var 2 B) (ArraySection (Var 2 A) [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 3 (Integer 4 [])))] (Integer 4 [((IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 5 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 4 (Integer 4 []))))]) ()) ())] Source Public Implementation () .false. .false.)}) [])
